From fe41204dedba2afd425b08e2306d49540c5f70df Mon Sep 17 00:00:00 2001
From: Daniel Baluta <daniel.baluta@nxp.com>
Date: Wed, 5 Jul 2017 18:52:51 +0300
Subject: [PATCH 2080/5242] MLK-15317-6: ARM64: dts: Add support for MQS in
 im8xp

commit  2668a443e9aaf2f6e4579b1e6117cc7fa3c04966 from
https://source.codeaurora.org/external/imx/linux-imx.git

We introduce a new dts for MQS which is supported by
imx8qxp with debug base board v2.

Connection in debug board is:
	MQS_LEFT:	SEAF_B_G39
	MQS_RIGHT:	SEAF_B_G38

This is similar with imx8qm. Note that these pins are shared with SPDIF.

Signed-off-by: Daniel Baluta <daniel.baluta@nxp.com>
Reviewed-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../devicetree/bindings/sound/imx-audio-mqs.txt    |    2 +-
 arch/arm64/boot/dts/freescale/Makefile             |    3 +-
 .../dts/freescale/fsl-imx8qxp-lpddr4-arm2-mqs.dts  |   56 ++++++++++++++++++++
 3 files changed, 59 insertions(+), 2 deletions(-)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2-mqs.dts

diff --git a/Documentation/devicetree/bindings/sound/imx-audio-mqs.txt b/Documentation/devicetree/bindings/sound/imx-audio-mqs.txt
index c0195c2..10ae15a 100644
--- a/Documentation/devicetree/bindings/sound/imx-audio-mqs.txt
+++ b/Documentation/devicetree/bindings/sound/imx-audio-mqs.txt
@@ -1,7 +1,7 @@
 Freescale i.MX audio complex with mqs codec
 
 Required properties:
-- compatible : "fsl,imx-audio-mqs", "fsl,imx8qm-lpddr4-arm2-mqs".
+- compatible : "fsl,imx-audio-mqs", "fsl,imx8qm-lpddr4-arm2-mqs", "fsl,imx8qxp-lpddr4-arm2-mqs"
 - model : The user-visible name of this sound complex
 - cpu-dai : The phandle of the i.MX sai controller
 - audio-codec : The phandle of the mqs audio codec
diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index c584bf2..9a938f4 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -22,5 +22,6 @@ dtb-$(CONFIG_ARCH_FSL_IMX8QM) += fsl-imx8qm-lpddr4-arm2.dtb \
 dtb-$(CONFIG_ARCH_FSL_IMX8QXP) += fsl-imx8qxp-lpddr4-arm2.dtb \
 				  fsl-imx8qxp-lpddr4-arm2-enet2.dtb \
 				  fsl-imx8qxp-lpddr4-arm2-gpmi-nand.dtb \
-				  fsl-imx8qxp-lpddr4-arm2-it6263.dtb
+				  fsl-imx8qxp-lpddr4-arm2-it6263.dtb \
+				  fsl-imx8qxp-lpddr4-arm2-mqs.dtb
 dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += fsl-imx8mq-evk.dtb
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2-mqs.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2-mqs.dts
new file mode 100644
index 0000000..b2169bec
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2-mqs.dts
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8qxp-lpddr4-arm2.dts"
+
+/ {
+	sound-cs42888 {
+		status = "disabled";
+	};
+
+	sound-mqs {
+		compatible = "fsl,imx8qxp-lpddr4-arm2-mqs",
+				"fsl,imx-audio-mqs";
+		model = "mqs-audio";
+		cpu-dai = <&sai1>;
+		audio-codec = <&mqs>;
+		asrc-controller = <&asrc1>;
+	};
+};
+
+&esai0 {
+	status = "disabled";
+};
+
+&iomuxc {
+
+	imx8qxp-arm2 {
+		pinctrl_mqs: mqsgrp {
+			fsl,pins = <
+				SC_P_SPDIF0_TX_ADMA_MQS_L	0xc6000061
+				SC_P_SPDIF0_RX_ADMA_MQS_R	0xc6000061
+			>;
+		};
+	};
+};
+
+&mqs {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_mqs>;
+	status = "okay";
+};
+
+&sai1 {
+	status = "okay";
+};
-- 
1.7.9.5

